kola/tests/docker/docker.go

```diff
@@ -459,17 +470,75 @@ func dockerUserNoCaps(c cluster.TestCluster) {
 	}
 }
 
-// testDockerInfo test that docker info's output is as expected.  the expected
-// filesystem may be asserted as one of 'overlay', 'btrfs', 'devicemapper'
-// depending on how the machine was launched.
-func testDockerInfo(expectedFs string, c cluster.TestCluster) {
+// dockerContainerdRestart ensures containerd will restart if it dies. It tests that containerd is running,
+// kills it, the tests that it came back up.
+func dockerContainerdRestart(c cluster.TestCluster) {
+	m := c.Machines()[0]
+	// verify containerd came up
+	if !testContainerdUp(c) {
+		c.Fatalf("Containerd did not start")
+	}
+
+	// kill it and verify it went down
+	pid, err := m.SSH("pgrep containerd")
```

Comment:
`systemctl show containerd -p MainPID --value` is arguably more robust.

I'm personally always wary of `pgrep`.